### PR TITLE
code rewritting

### PR DESCRIPTION
**Pull request by Drulac**
No deep change, no function change

- simplifications/better code rewritting
- new comments
- variable rename for improved understanding
- smalls performance corrections

_sorry, my linter also make some little changes like a space who disapear somewhere, which add some lines changement for nothing :-/_

### CODE REVIEW

**Comment by Drulac** [comment]
les tests passaient tous sur mon PC :thinking:

Je n'arrive pas à trouver dans l'interface de Travis la source du problème :sweat_smile:

File: packages/remark-grid-tables/src/index.js [hunk 1]
```diff
@@ -542,7 +384,12 @@ function createGrid (nbRows, nbCols) {
   for (let i = 0; i < nbRows; i++) {
     grid.push([])
     for (let j = 0; j < nbCols; j++) {
-      grid[i].push({height: -1, width: -1, hasBottom: true, hasRigth: true})
+      grid[i].push({
+        height: -1,
+        width: -1,
+        hasBottom: true,
+        hasRigth: true,
+      })
```
Comment (TitiAlone) [inline packages/remark-grid-tables/src/index.js:392]: This one is good. I would change the `hasRigth` for the correct verison of it.

File: packages/remark-grid-tables/src/index.js [hunk 2]
```diff
@@ -386,154 +285,97 @@ function generateTable (tableContent, now, tokenizer) {
...
+    (part, partIndex) => {
+      const partElt = {
+        type: 'tableHeader',
         children: [],
         data: {
-          hName: 'tr',
+          hName:
+            hasHeader && partIndex === 0
+              ? 'thead'
+              : 'tbody',
```
Comment (TitiAlone) [inline packages/remark-grid-tables/src/index.js:299]: Okish change, but still prefer to keep all on one line

File: packages/remark-grid-tables/src/index.js [hunk 3]
```diff
@@ -552,93 +399,136 @@ function createGrid (nbRows, nbCols) {
...
   }
 }
 
 function extractAST (gridNode, grid) {
   let i = 0
   /* Fill the grid with value, height and width from the ast */
-  gridNode.children.forEach(th => {
-    th.children.forEach(row => {
+  gridNode.children.forEach((th) => {
+    th.children.forEach((row) => {
```
Comment (TitiAlone) [inline packages/remark-grid-tables/src/index.js:435]: No! Why would you wrap these in parens? It works perfectly fine without.

File: packages/remark-grid-tables/src/index.js [hunk 4]
```diff
@@ -552,93 +399,136 @@ function createGrid (nbRows, nbCols) {
...
+        for (
+          let x = 0;
+          x < cell.data.hProperties.rowspan;
+          x++
+        ) {
+          for (
+            let y = 0;
+            y < cell.data.hProperties.colspan;
+            y++
+          ) {
```
Comment (TitiAlone) [inline packages/remark-grid-tables/src/index.js:471]: Single line

File: packages/remark-grid-tables/src/index.js [hunk 5]
```diff
@@ -758,14 +662,268 @@ function stringifyGridTables (gridNode) {
   return gridString.join('\n')
 }
 
+/*
+ *
+ * Reviewed
+ *
+ */
```
Comment (TitiAlone) [inline packages/remark-grid-tables/src/index.js:669]: I would say remove this comment

**Comment by Drulac** [comment]
Wow, long review :-) !

All the added breaks are from my linter (prettier)

I will take a look at it in the week or in aout

**Comment by TitiAlone** [comment]
> All the added breaks are from my linter (prettier)

Well, that's the problem, then; you should use eslint with the configuration file used in the project (`eslintrc` at root).

**Comment by TitiAlone** [comment]
Hi, any updates here?

**Comment by Drulac** [comment]
I've just came back from holidays, I will do it in days

**Comment by arobase-che** [comment]
Don't please. The refactoring is promising.

**Comment by TitiAlone** [comment]
Depends if you think you'll have time later?

**Comment by arobase-che** [comment]
Can we re-open a closed issue ?

Because i will not for a while.

**Comment by TitiAlone** [comment]
We can reopen it, but I'm afraid the code may have changed in time...

**Comment by arobase-che** [comment]
Do it if you want. I don't want to commit myself to it.

**Comment by TitiAlone** [comment]
I'm closing the PR for now, if someone wants to take a look at it, I can reopen anytime.

**Pull request closed without merging**